pkg/diagnostics/types/flags.go

```diff
@@ -0,0 +1,53 @@
+package types
+
+import (
+	"github.com/openshift/origin/pkg/cmd/server/start"
+	flag "github.com/spf13/pflag"
+	"strings"
+)
+
+type Flags struct {
+	Diagnostics    List // named diagnostics to run
+	OpenshiftPath  string
+	OscPath        string
+	LogLevel       int
+	Format         string          // format of output - text/json/yaml
+	OpenshiftFlags *flag.FlagSet   // flags that were set on the command run
+	CanCheck       map[Target]bool // components to diagnose - master/node/client
+	MustCheck      Target
+	// for diagnosing "all", enable specifying config file locations
+	ClientConfigPath string
+	MasterConfigPath string
+	NodeConfigPath   string
+	// the master has its own container for flags/options
+	MasterOptions *start.MasterOptions
+	// the node has its own container for flags/options
+	NodeOptions *start.NodeOptions
```

Comment:
The interplay is a little complicated. The same command can be checking any/all of master,node,client. Some options are shared between those (e.g. syslogd unit status between node/master; list of named diagnostics you want to run), some aren't.

I think it will probably be possible/useful to more clearly delineate these as separate objects as you say. At a minimum, one for handling the top-level "what did the user say and what is discoverable" and one each for handing off to actually run master,node,client diagnostics.